### PR TITLE
Home w26.1

### PR DESCRIPTION
**Pull request by KostiantynVorobiov**

### CODE REVIEW

File: src/main/webapp/WEB-INF/views/product/add.jsp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Add new product</title>
+</head>
+<body>
+
+<form method="post" action="${pageContext.request.contextPath}/products/add">
+    Product name: <input type="text" name="name">
+    Price: <input type="text" name="price">
```
Comment (Vosker) [inline src/main/webapp/WEB-INF/views/product/add.jsp:10]: сюда можешь поставить type="number"

File: src/main/webapp/WEB-INF/views/product/add.jsp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    <button type="submit">Add</button>
+</form>
+<a href="${pageContext.request.contextPath}/">Home</a>
+</body>
```
Comment (Vosker) [inline src/main/webapp/WEB-INF/views/product/add.jsp:15]: Было бы неплохо добавить еще ссылку на возврат ко всем продуктам.

**Review by Vosker — COMMENTED** [review]
Good job!

File: src/main/java/com/internet/shop/controller/DeleteUserController.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
+        Long id = Long.valueOf(userId);
+        userService.deleteById(id);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/DeleteUserController.java:23]: let's delete cart here as well

File: src/main/java/com/internet/shop/controller/GetAllProductController.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllProductController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/GetAllProductController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UserController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/UserController.java]:
```suggestion
public class GetAllUsersController extends HttpServlet {
```

File: src/main/webapp/WEB-INF/views/injectData.jsp [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Demo</title>
+</head>
+<body>
+<h1>Your mock data was added to DB</h1>
+
+
```
Comment (RomanDubovskyi) [inline src/main/webapp/WEB-INF/views/injectData.jsp]:
```suggestion
```

File: src/main/webapp/web.xml [hunk 7]
```diff
@@ -3,14 +3,90 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>cartProducts</servlet-name>
+        <servlet-class>com.internet.shop.controller.GetShoppingCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>cartProducts</servlet-name>
+        <url-pattern>/shopping-cart/cart</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]: fix url patterns everywhere according to example from checklist 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/com/internet/shop/controller/GetAllProductController.java (hunk 4 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/controller/GetAllProductController.java]: Done

File: src/main/java/com/internet/shop/controller/UserController.java (hunk 5 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/controller/UserController.java]: Done

File: src/main/webapp/WEB-INF/views/injectData.jsp (hunk 6 above)
Comment (KostiantynVorobiov) [reply src/main/webapp/WEB-INF/views/injectData.jsp]: Done

File: src/main/java/com/internet/shop/controller/DeleteUserController.java (hunk 3 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/controller/DeleteUserController.java:23]: Done. I hope a did it correctly)

File: src/main/webapp/web.xml (hunk 7 above)
Comment (KostiantynVorobiov) [reply src/main/webapp/web.xml]: Done

File: src/main/java/com/internet/shop/controller/DeleteUserController.java [hunk 8]
```diff
@@ -11,13 +12,16 @@
...
+    private final ShoppingCartService shoppingCartService = (ShoppingCartService) injector
+            .getInstance(ShoppingCartService.class);
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         String userId = req.getParameter("id");
         Long id = Long.valueOf(userId);
         userService.deleteById(id);
+        shoppingCartService.deleteById(id);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controller/DeleteUserController.java]: you should pass shoppingCartId but you're passing userId

Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/controller/DeleteUserController.java]: Understand. Done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**